packages/types/src/extrinsic/ExtrinsicPayload.ts

```diff
@@ -36,6 +36,24 @@ function decodeExtrinsicPayload (registry: Registry, value?: GenericExtrinsicPay
     return value.unwrap();
   }
 
+  /**
+   * HACK: In order to change the assetId from `number | object` to HexString (While maintaining the true type ie Option<TAssetConversion>),
+   * to allow for easier generalization of the SignerPayloadJSON interface the below check is necessary. The ExtrinsicPayloadV4 class does not like
+   * a value passed in as an Option, and can't decode it properly. Therefore, we ensure to convert the following below, and then pass the option as a unwrapped
+   * JSON value.
+   *
+   * ref: https://github.com/polkadot-js/api/pull/5968
+   * ref: https://github.com/polkadot-js/api/pull/5967
+   */
+  if (value && (value as ExtrinsicPayloadValue).assetId && isHex((value as ExtrinsicPayloadValue).assetId)) {
+    const adjustedPayload = {
+      ...(value as ExtrinsicPayloadValue),
+      assetId: registry.createType('TAssetConversion', hexToU8a((value as ExtrinsicPayloadValue).assetId)).toJSON()
```

Comment:
Is passing it as `JSON` necessary? I can't seem to find why can't we just use the `TAssetConversion`

Reply:
Let me check this one, yea we might be able to just pass in the main type!

Reply:
Ahh yes, the reason I set it to JSON was because the `ExtrinsicPayloadValue` doesn't take any Codec values to begin with, so to align with the current interface, I used a decoded value!

Reply:
Makes sense